core/src/main/java/io/undertow/server/ConnectorStatisticsImpl.java

```diff
@@ -71,42 +71,42 @@ public void exchangeEvent(HttpServerExchange exchange, NextListener nextListener
 
     @Override
     public long getRequestCount() {
-        return requestCountUpdater.get(this);
+        return requestCount;
     }
 
     @Override
     public long getBytesSent() {
-        return bytesSentUpdater.get(this);
+        return bytesSent;
     }
 
     @Override
     public long getBytesReceived() {
-        return bytesReceivedUpdater.get(this);
+        return bytesReceived;
     }
 
     @Override
     public long getErrorCount() {
-        return errorCountUpdater.get(this);
+        return errorCount;
     }
 
     @Override
     public long getProcessingTime() {
-        return processingTimeUpdater.get(this);
+        return processingTime;
     }
 
     @Override
     public long getMaxProcessingTime() {
-        return maxProcessingTimeUpdater.get(this);
+        return maxProcessingTime;
     }
 
     @Override
     public void reset() {
-        requestCountUpdater.set(this, 0);
-        bytesSentUpdater.set(this, 0);
-        bytesReceivedUpdater.set(this, 0);
-        errorCountUpdater.set(this, 0);
-        maxProcessingTimeUpdater.set(this, 0);
-        processingTimeUpdater.set(this, 0);
+        requestCount = 0;
```

Comment:
These need to be set() invocations. From the JavaDoc:

"Note that the guarantees of the {@code compareAndSet}
 * method in this class are weaker than in other atomic classes.
 * Because this class cannot ensure that all uses of the field
 * are appropriate for purposes of atomic access, it can
 * guarantee atomicity only with respect to other invocations of
 * {@code compareAndSet} and {@code set} on the same updater."

This means that if reset() is called and statistics are in the process of being updated the 0 value that is set here may be overwritten by the old value.